### PR TITLE
Decouple Util and StringUtil

### PR DESCRIPTION
**Pull request by denizt**
Util included StringUtil which lead to many classes relying on it, decouple it instead and make it explicit to include StringUtils.

This was achieved by moving the sortstringbyname from Util class to a better fit, StringUtil. The method was also optimized to avoid copying of the string and use the method from the StringUtil instead.

### CODE REVIEW

**Comment by Montellese** [comment]
I'm always in favour of untangling utility classes/implementations.

**Comment by Paxxi** [comment]
:+1:

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by topfs2** [comment]
Nice, cleanups is always welcomed

File: xbmc/utils/StringUtils.h [hunk 1]
```diff
@@ -194,3 +194,11 @@ class StringUtils
 private:
   static std::string m_lastUUID;
 };
+
+struct sortstringbyname
+{
+  bool operator()(const std::string& strItem1, const std::string& strItem2)
+  {
+    return StringUtils::CompareNoCase(strItem1, strItem1) < 0;
```
Comment (tobbi) [inline xbmc/utils/StringUtils.h]: The second one should be strItem2, right? Otherwise you're comparing strItem1 with itself.

**Comment by denizt** [comment]
thx @tobbi, copy paste fail. Fixed

**Comment by mkortstiege** [comment]
jenkins build this please

**Comment by denizt** [comment]
@mkortstiege I wrote a unit test for this, how do you usually do this? Same commit or pre/post this commit?

**Comment by mkortstiege** [comment]
There's no hard rule. I'd add it as separate commit to this one.

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by MartijnKaijser** [comment]
@popcornmix is that build failure related? Doesn't look like it

**Comment by popcornmix** [comment]
The build log seems to have been deleted:
http://jenkins.kodi.tv/job/LINUX-RBPI/3157/

**Comment by MartijnKaijser** [comment]
let's try again
jenkins build this please

**Comment by Montellese** [comment]
@popcornmix: Failed again on RPI with
```
CPP     xbmc/cores/VideoRenderers/MMALRenderer.o
In file included from MMALRenderer.cpp:22:0:
MMALRenderer.h:127:27: fatal error: LinuxRenderer.h: No such file or directory
compilation terminated.
make[1]: *** [MMALRenderer.o] Error 1
make: *** [xbmc/cores/VideoRenderers/VideoRenderer.a] Error 2
```
Doesn't look related to this PR but since it failed twice now on RPI it would probably be good to look into it.

**Comment by popcornmix** [comment]
I'm a bit confused why several Pi builds have succeeded today (http://jenkins.kodi.tv/job/LINUX-RBPI/).
The problem is that HAS_MMAL is not defined in this PR's build. It should always be defined for Pi builds (and usually is).

**Comment by mkortstiege** [comment]
Regardless of HAS_MMAL.. shouldn't it be `LinuxRendererGL.h` or the GLES variant?

**Comment by popcornmix** [comment]
Cherry-picking this PR does break my Pi build too.
Looks like MMALRenderer.h is not explicitly including "system.h", but currently getting it though some convoluted chain, that this PR breaks.

Let me submit a PR that fixes that. I think the whole idef HAS_MMAL in header file is unnecessary - it doesn't seem to be included except when HAS_MMAL should be set.

**Comment by popcornmix** [comment]
Hopefully will be fixed with: https://github.com/xbmc/xbmc/pull/6588

**Comment by popcornmix** [comment]
jenkins build this please

**Comment by MartijnKaijser** [comment]
jenkins build this please, just do it

**Pull request merged**